driver/src/main/java/org/neo4j/driver/v1/Config.java

```diff
@@ -241,32 +241,44 @@ public Config toConfig()
     }
 
     /**
-     * A configuration to configure TLS authentication
+     * Control the level of encryption to require
+     */
+    public enum EncryptionLevel
+    {
+        /** With this level, the driver will only connect to the server if it can do it without encryption. */
+        REJECTED,
```

Comment:
The point of REQUIRED/REJECTED was that it allows adding OPTIONAL later, so if you're connecting to a server that may or may not require TLS, the driver won't care. This is a nice option if you want to centralize control of this policy - you only have to configure the server to say `REQUIRE`, and as long as all the drivers are set to `OPTIONAL` they will follow suit.

Without that option it becomes hard, I think, to change the policy without full system downtime as well. This is how Oracle defines their encryption policy.

Obviously, we don't have OPTIONAL on the driver side here, but the naming of the other two was intended to leave a slot for that in a later version.